### PR TITLE
[OptimizeInstructions] More canonizations for floating points

### PR DESCRIPTION
**Pull request by MaxGraey**
```rust
x - C       ->    x + (-C)
min(C, x)   ->    min(x, C)
max(C, x)   ->    max(x, C)
```
And remove redundant rules

### CODE REVIEW

**Comment by MaxGraey** [comment]
@kripken could you take a look please?

File: src/passes/OptimizeInstructions.cpp [hunk 1]
```diff
@@ -3507,20 +3507,6 @@ struct OptimizeInstructions
...
     }
-    {
-      double value;
-      if (matches(curr, binary(Sub, any(), fval(&value))) && value == 0.0) {
-        // x - (-0.0)   ==>   x + 0.0
-        if (std::signbit(value)) {
-          curr->op = Abstract::getBinary(type, Add);
-          right->value = right->value.neg();
-          return curr;
-        } else if (fastMath) {
```
Comment (kripken) [inline src/passes/OptimizeInstructions.cpp:3518]: Don't we still want the fast-math part?

Comment (MaxGraey) [reply src/passes/OptimizeInstructions.cpp:3518]: This redundant rule. FastMath part already handled here:
https://github.com/WebAssembly/binaryen/blob/ca6a212b5a687fd5d53d222189423915f7fb1778/src/passes/OptimizeInstructions.cpp#L3526

Comment (kripken) [reply src/passes/OptimizeInstructions.cpp:3518]: I see, thanks!

**Review by kripken — APPROVED** [review]
lgtm after fuzzing.

**Comment by MaxGraey** [comment]
Fuzzed `ITERATION: 4450`

**Pull request merged**